### PR TITLE
Track dependencies when using qAddDependentFile

### PR DESCRIPTION
**Pull request by mpickering**
Do I also need to `need` the files or is calling `doesFileExist` enough to track the dependency?

### CODE REVIEW

**Comment by ndmitchell** [comment]
`doesFileExist` tracks that the file exists or not. `need` tracks the modification time too and asserts that it must exist. My guess is you want `need` here?

**Comment by pepeiborra** [comment]
`need` uses the plain Shake file rules, wouldn't `GetModificationTime` be a better fit here?

**Comment by mpickering** [comment]
@pepeiborra What is your exact suggestion?

**Comment by pepeiborra** [comment]
`use_ GetModificationTime` instead of `doesFileExist`

**Comment by ndmitchell** [comment]
Yes, you're right. A `GetModificationTime` would be better.

**Comment by mpickering** [comment]
Is this what you have in mind?

**Comment by pepeiborra** [comment]
> Is this what you have in mind?

Yes, this is exactly what I had in mind

**Comment by mpickering** [comment]
How would a test work?

**Comment by pepeiborra** [comment]
Check that a program using `qAddDependentFile` to e.g. inject code, gets recompiled after changing the indirect dep?

**Comment by cocreature** [comment]
a very simple way to check that it gets recompiled might be to just delete the file that you are adding via `qAddDependentFile` and see that you get a diagnostic afterwards.

**Comment by mpickering** [comment]
I added a test now.

File: test/exe/Main.hs [hunk 1]
```diff
@@ -2009,6 +2010,42 @@ loadCradleOnlyonce = testGroup "load cradle only once"
...
+              , "module Foo where"
+              , "import Language.Haskell.TH.Syntax"
+              , "foo :: Int"
+              , "foo = 1 + $(do"
+              , "               qAddDependentFile \"dep-file.txt\""
+              , "               f <- qRunIO (readFile \"dep-file.txt\")"
+              , "               if f == \"B\" then [| 1 |] else lift f)"
+              ]
+        let bazContent = T.unlines ["module Baz where", "import Foo"]
+        _ <- openDoc' "Foo.hs" "haskell" fooContent
```
Comment (cocreature) [inline test/exe/Main.hs:2033]:
```suggestion
        _ <-createDoc "Foo.hs" "haskell" fooContent
```

File: test/exe/Main.hs [hunk 2]
```diff
@@ -2009,6 +2010,42 @@ loadCradleOnlyonce = testGroup "load cradle only once"
...
+        doc <- openDoc' "Baz.hs" "haskell" bazContent
```
Comment (cocreature) [inline test/exe/Main.hs:2034]:
```suggestion
        doc <- createDoc "Baz.hs" "haskell" bazContent
```

**Pull request merged**